Sources/_StringProcessing/Regex/Match.swift

```diff
@@ -40,6 +40,9 @@ extension Regex.Match {
     let typeErasedMatch = anyRegexOutput.existentialOutput(
       from: anyRegexOutput.input
     )
+    guard typeErasedMatch is Output else {
+      fatalError()
+    }
```

Comment:
This seems like it's doing the same thing as the `as!`. Do we want a custom error message with the fatal error? Could rewrite as `guard let output = typeErasedMatch...` if so.